### PR TITLE
[wptpr.live] Make sure to clean up old refs/prs-open/{pr}

### LINKED ISSUE #36: pr-preview sychronize fails to handle pull requests outside its window
The [pr-preview script in WPT](https://github.com/web-platform-tests/wpt/blob/master/tools/ci/pr_preview.py) is responsible for determining what pull requests should be mirrored. It runs every 5 minutes on a cronjob, and looks at an 8 minute window of changed pull requests to decide what to do.

The problem with this is that if pr-preview ever breaks or doesn't run in some 8 minute window, for whatever reason, then PRs are missed. This can miss both PRs that are closed (so should have their deployment deleted) as well as PRs that are opened (and just won't get a deployment). The former is a larger problem than the latter.

Right now, there are 103 PRs with a `refs/prs-open/{pr}` ref for them. Of those, only 59 are open and 44 are closed. At ~400MB a sync, that's 18GB of unnecessary disk space! (see https://github.com/web-platform-tests/wpt.live/issues/35).

It's not feasible to scan all PRs, but what we can do is scan the existing refs (along with the window) and check if any should be updated or deleted. This will definitely catch deletions, but could still miss additions - however those can be trivially fixed by anyone commenting on the pull request or close/re-opening it, so it seems more reasonable to miss them.

### PR DESCRIPTION
**Pull request by stephenmcgruer**
This expands the list of PRs that pr_preview.py looks at to include
those that have existing refs. That means that old PRs outside the
window will always have their deployments deleted, rather than being
kept forever.

Fixes https://github.com/web-platform-tests/wpt.live/issues/36

### CODE REVIEW

File: tools/ci/pr_preview.py [hunk 1]
```diff
@@ -121,20 +130,22 @@ def get_pull_requests(self, updated_since):
...
         logger.info('Checking if pull request %s is from a fork', pr_number)
 
-        data = gh_request('GET', url)
+        # We may already have this information; if so no need to make another
+        # API call.
+        if 'head' not in pull_request:
+            url = '{}/repos/{}/pulls/{}'.format(
+                self._host, self._github_project, pr_number
+            )
+            pull_request = gh_request('GET', url)
```
Comment (Hexcles) [inline tools/ci/pr_preview.py:146]: Can we call `get_pull_request` directly?

```suggestion
            pull_request = self.get_pull_request(pr_number)
```

File: tools/ci/pr_preview.py [hunk 2]
```diff
@@ -256,6 +267,32 @@ def delete_ref(self, refspec):
             full_ref
         ])
 
+    def get_pr_open_numbers(self):
```
Comment (Hexcles) [inline tools/ci/pr_preview.py:270]: It looks like `pr` is rarely, if ever, used in public method names. For consistency, maybe `get_open_pull_requests` and document the return value is a list of numbers?

File: tools/ci/pr_preview.py [hunk 3]
```diff
@@ -256,6 +267,32 @@ def delete_ref(self, refspec):
...
+        refspec = 'refs/prs-open/*'
+
+        logger.info('Fetching all ref names matching "%s"', refspec)
+
+        output = subprocess.check_output([
+            'git',
+            '-c',
+            'credential.username={}'.format(self._token),
+            '-c',
+            'core.askPass=true',
```
Comment (Hexcles) [inline tools/ci/pr_preview.py:282]: We now have at least three of these. Maybe create a helper method?

**Review by Hexcles — APPROVED** [review]
LGTM % nits

File: tools/ci/pr_preview.py (hunk 2 above)
Comment (stephenmcgruer) [reply tools/ci/pr_preview.py:270]: Yeah, I struggled to convey this is only looking at the refs in the remote, and doesn't touch github. Switched to `get_pull_requests_with_open_ref`, wdyt?

File: tools/ci/pr_preview.py (hunk 3 above)
Comment (stephenmcgruer) [reply tools/ci/pr_preview.py:282]: Moved the joint `git -c ... -c ...` part into a helper.

File: tools/ci/pr_preview.py (hunk 2 above)
Comment (Hexcles) [reply tools/ci/pr_preview.py]: SGTM

**Pull request merged**